### PR TITLE
Fix dynamic load score calculation

### PR DESCRIPTION
**Pull request by alekece**
### Description
This PR aims to deal with a bug when we dynamically update the node `load_score`.

### How was this PR tested?
Tested locally:
1. Started locally a node writer with some pre-existing shards 
2. Check initial load score and number of shards
3. Create shard using gRPC method `NewShard`
4. Add paragraph using gRPC method `SetResource`
5. Delete shard using gRPC method `DeleteShard`

### CODE REVIEW

File: nucliadb_node/src/bin/writer.rs [hunk 1]
```diff
@@ -234,8 +241,7 @@ pub fn get_node_metadata() -> NodeWriterMetadata {
 
         match shard.get_info(&GetShardRequest::default()) {
             Ok(count) => {
-                node_metadata.shard_count += 1;
-                node_metadata.paragraph_count += count.paragraphs as u64;
+                node_metadata.new_shard(shard.id, count.paragraphs as u64);
```
Comment (hermeGarcia) [inline nucliadb_node/src/bin/writer.rs:244]: can be done without a block:
```rust
Ok(count) => node_metadata.new_shard(shard.id, count.paragraphs as u64),
```

File: nucliadb_node/src/bin/writer.rs [hunk 2]
```diff
@@ -147,25 +147,32 @@ pub async fn watch_node_update(
...
+
+        if let Some(load_score) = load_score {
+            info!("Update node state: load_score = {load_score}");
+            node.update_state(LOAD_SCORE_KEY, load_score).await;
+        }
+
+        if let Some(shard_count) = shard_count {
+            info!("Update node state: shard_count = {shard_count}");
+            node.update_state(SHARD_COUNT_KEY, shard_count).await;
         }
```
Comment (hermeGarcia) [inline nucliadb_node/src/bin/writer.rs:176]: Maybe the following approach would simplify your loop:
```rust
async fn update_node_state<T: std::fmt::Display>(node: &NodeHandle, key: Key<T>, score: T) {
    info!("Update node state: {} = {score}", key.as_str());
    node.update_state(key, score).await;
}

pub async fn watch_node_update(
    node: NodeHandle,
    mut receiver: UnboundedReceiver<NodeWriterEvent>,
    mut metadata: NodeWriterMetadata,
) {
    info!("Start node update task");

    while let Some(event) = receiver.recv().await {
        match event {
            NodeWriterEvent::ShardCreation(id) => {
                metadata.new_empty_shard(id);
                update_node_state(&node, SHARD_COUNT_KEY, metadata.shard_count()).await;
            }
            NodeWriterEvent::ShardDeletion(id) => {
                metadata.delete_shard(id);
                update_node_state(&node, LOAD_SCORE_KEY, metadata.load_score()).await;
                update_node_state(&node, SHARD_COUNT_KEY, metadata.shard_count()).await;
            }
            NodeWriterEvent::ParagraphCount(id, paragraph_count) => {
                metadata.update_shard(id, paragraph_count);
                update_node_state(&node, LOAD_SCORE_KEY, metadata.load_score()).await;
            }
        }
    }

    info!("Node update task stopped");
}
```

**Review by hermeGarcia — APPROVED** [review]
Just code style comments

File: nucliadb_node/src/bin/writer.rs (hunk 2 above)
Comment (alekece) [reply nucliadb_node/src/bin/writer.rs:176]: Nice suggestion :cat:

File: nucliadb_node/src/bin/writer.rs (hunk 1 above)
Comment (alekece) [reply nucliadb_node/src/bin/writer.rs]: Done 807bf855cfe52417caf208cb76ee1b30f2c7598c

File: nucliadb_node/src/bin/writer.rs (hunk 2 above)
Comment (alekece) [reply nucliadb_node/src/bin/writer.rs]: Done at 807bf855cfe52417caf208cb76ee1b30f2c7598c

**Review by hermeGarcia — APPROVED** [review]

**Pull request merged**